### PR TITLE
Bring ltimes benchmark up to date

### PR DESCRIPTION
**Pull request by artv3**
# Summary

This PR bring the ltimes kernel up to date. It is currently a WIP.

### CODE REVIEW

File: include/RAJA/pattern/tensor/TensorIndex.hpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ namespace expt
 {
 
 
-  template<typename IDX, typename TENSOR_TYPE, camp::idx_t DIM, IDX INDEX_VALUE, strip_index_type_t<IDX> LENGTH_VALUE>
```
Comment (rchen20) [inline include/RAJA/pattern/tensor/TensorIndex.hpp:32]: Support for strongly typed indices with `TensorIndex` was missing, and that is fixed here. We don't have this in our CI testing, so I'll create an issue to add tests for that.

File: scripts/lc-builds/toss4_amdclang.sh [hunk 2]
```diff
@@ -73,6 +73,7 @@ cmake \
   -DENABLE_HIP=ON \
   -DENABLE_OPENMP=ON \
   -DENABLE_CUDA=OFF \
+  -DENABLE_BENCHMARKS=ON \
```
Comment (rhornung67) [inline scripts/lc-builds/toss4_amdclang.sh:76]: Should we make this sort of thing an argument you can pass to the script like we do for compiler version, compute architecture, etc?

**Review by rhornung67 — APPROVED** [review]

**Review by rchen20 — APPROVED** [review]

File: scripts/lc-builds/blueos_nvcc_gcc.sh [hunk 3]
```diff
@@ -9,22 +9,30 @@
...
-BUILD_SUFFIX=lc_blueos-nvcc${COMP_NVCC_VER}-${COMP_ARCH}-gcc${COMP_GCC_VER}
+if [[ $# -gt 3 ]]; then 
+  BENCHMARKS=$4
+  shift 4
+  BUILD_SUFFIX=lc_blueos-nvcc${COMP_NVCC_VER}-${COMP_ARCH}-gcc${COMP_GCC_VER}-benchmark_${BENCHMARKS}
+else
+  shift 3
+  BENCHMARKS=Off
+  BUILD_SUFFIX=lc_blueos-nvcc${COMP_NVCC_VER}-${COMP_ARCH}-gcc${COMP_GCC_VER}
+fi
```
[the remark this answers is not in the document]
Comment (rhornung67) [reply scripts/lc-builds/blueos_nvcc_gcc.sh:35]: That's fine, but I don't think it's necessary. We only have a few benchmark problems, so building them is not a big deal. We could just enable benchmarks in all our scripts and folks can turn them off easily enough if they want.  We ill enable them in a subset of CI builds.

[the remark this answers is not in the document]
Comment (rhornung67) [reply scripts/lc-builds/blueos_nvcc_gcc.sh:35]: I don't think a command line option is needed. We have to edit the build script files to enable/disable features, back-ends, etc. Having a special command line arg for this one case seems confusing and less convenient for me.

[the remark this answers is not in the document]
Comment (artv3) [reply scripts/lc-builds/blueos_nvcc_gcc.sh:35]: Okay, I can revert and just turn them on across the board. Is this right?

**Comment by artv3** [comment]
I think this is good to go now.

**Review by rhornung67 — APPROVED** [review]

**Comment by rchen20** [comment]
@artv3 Did we want to turn on building `ltimes` in `benchmark/CMakeLists.txt`?

**Comment by rhornung67** [comment]
@artv3 if you don't need/want the branch from this PR, please delete. Thank you.

**Pull request merged**